c_glib/test/test-fixed-size-list-data-type.rb

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestFixedSizeListDataType < Test::Unit::TestCase
+  sub_test_case(".new") do
+    def setup
+      @value_type = Arrow::BooleanDataType.new
+      @list_size = 5
+      @field_name = "bool_field"
+    end
+
+    def test_field
+      field = Arrow::Field.new(@field_name, @value_type)
+      data_type = Arrow::FixedSizeListDataType.new(field, @list_size);
+      assert_equal(Arrow::Type::FIXED_SIZE_LIST, data_type.id);
```

Comment:
We should also check `data_type.field`.

Reply:
What do you think this change?

```diff
+      data_type = Arrow::FixedSizeListDataType.new(field, @list_size)
+      assert_equal([Arrow::Type::FIXED_SIZE_LIST, "fixed_size_list<bool_field: bool>[5]"],
+                   [data_type.id, data_type.to_s])
```

We can check size: 5, fileld_name: `bool_field`, and type: `bool`. 

It seems `FixedSizeListDataType` class can't get size and type separately.

https://github.com/apache/arrow/blob/57439e08cacf842ac2ce44bc72cf67196d4c184b/cpp/src/arrow/type_test.cc#L1680-L1698

Reply:
We can use this PR.